test/backend/certificates.spec.js

```diff
@@ -0,0 +1,88 @@
+import {assert, expect} from 'chai';
+import fs from 'fs';
+import {contains, dissoc, merge} from 'ramda';
+
+import {getSetting, saveSetting} from '../../backend/settings.js';
+import {
+    getCerts,
+    CA_HOST_URL,
+    calculateDaysToRenewal,
+    fetchCertsFromCA,
+    saveCertsLocally,
+    setRenewalJob,
+    renewCertificate
+} from '../../backend/certificates';
+import {username, accessToken} from './utils';
+
+const cleanUp = () => {
+    ['KEY_FILE', 'CERT_FILE'].forEach(fileName => {
+        try {
+            fs.unlinkSync(getSetting(fileName));
+        } catch (e) {}
+    });
+    try {
+        fs.unlinkSync(getSetting('SETTINGS_PATH'));
+    } catch (e) {}
+};
+
+describe('Certificates', function() {
+    beforeEach(() => {
+        cleanUp();
+    });
+
+    afterEach(() => {
+        cleanUp();
+    });
+
+    it('The CA URL is correctly defined.', () => {
+        assert.equal(CA_HOST_URL, 'http://plotly-connector-test.com/certificate');
+    });
+
+    it('Can populate relevant certs files to local storage path.', () => {
+        ['KEY_FILE', 'CERT_FILE'].forEach(fileName => {
+            assert.isFalse(
+                fs.existsSync(getSetting(fileName)),
+                `${fileName} did not exist at first`
+            );
+        });
+
+        const json = {key: 'key', cert: 'cert', subdomain: 'subdomain'};
+        const expectedKey = 'key';
+        const expectedCert = 'cert';
+        const expectedHost = 'subdomain.plotly-connector-test.com';
+
+        const response = saveCertsLocally(json);
+
+        assert.isNotOk(response.error);
```

Comment:
we don't use this `response` pattern anywhere else in the codebase I don't think. It looks like `saveCertsLocally` just returns `{}` anyway. Could we handle exceptions by throwing errors instead of returning error messages?